src/nunavut/lang/c/__init__.py

```diff
@@ -268,7 +268,7 @@ class _CFit(enum.Enum):
     IN_64 = 64
 
     def to_std_int(self, is_signed: bool) -> str:
-        return "{}int{}_t".format(("" if is_signed else "u"), self.value)
+        return "{}int_fast{}_t".format(("" if is_signed else "u"), self.value)
```

Comment:
> we shouldn't change the Nunavut ABI contract

Did I understand correctly that the binary layout should be preserved for deserialized structures? Not just serialized.

If you use uint_leastXX_t instead of uint_fastXX_t, then on machines with CHAR_BIT==8 (the only ones you had before), the packaging of the fields will not change. Your customers will not be harmed. Switch to uint_leastXX_t?

Reply:
Right, for a given set of Nunavut configuration options and for the same DSDL inputs the type code generated should compile to be bit-for-bit the same as it is today. That is, the binaries that include ser/des only need to maintain interface and behavioural contracts but the structs must be bit-for-bit the same. You can test this easily by generating before and after with serialization disabled (will just generate PoCos). If we change from int to fast_int we are going to modify the alignment of some types on some platforms. This is not acceptable without a major revision to the Nunavut version number which we are not keen to do right now.

Reply:
However, using uint_leastXX_t instead of uint_fastXX_t will not change the alignment of the structures. Is that not enough?

Reply:
Can we do something like
```
def to_std_int(self, is_signed: bool, optimization: typing.Optional[str]) -> str:
    if optimization is not None:
        if optimization == "fast":
...
        elif optimization == "small":
...
```

etc. Then you can provide these hints only where you have non-CHAR_BIT==8 code. Is that possible?

Reply:
A lot of settings is also bad. Maybe it has sense to consider this possibility later.

In addition, all compilers, that I have deal with, define the types (u)int_fastXX_t the same as (u)int_leastXX_t. So, in most cases we don't lose anything, using (u)int_leastXX_t. Rather, the type (u)int_fastXX_t is provided for potential optimization